services/src/main/java/org/apache/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -751,14 +791,32 @@ public void onFailure(Response response, Throwable failure)
       super.onFailure(response, failure);
     }
 
-    private void emitQueryTime(long requestTimeNs, boolean success)
+    private void emitQueryTime(long requestTimeNs, boolean success, @Nullable String sqlQueryId)
     {
-      QueryMetrics queryMetrics = DruidMetrics.makeRequestMetrics(
-          queryMetricsFactory,
-          warehouse.getToolChest(query),
-          query,
-          req.getRemoteAddr()
-      );
+      QueryMetrics queryMetrics;
+      if (sqlQueryId != null) {
+        queryMetrics = queryMetricsFactory.makeMetrics();
+        if (queryMetrics == null) {
+          return;
+        }
+        queryMetrics.remoteAddress(req.getRemoteAddr());
+        // Setting sqlQueryId dimension to the metric. Using a dummy query since SQL is translated to a native query
+        // only at a broker.
+        queryMetrics.sqlQueryId(
```

Comment:
the implementations only need the sqlQueryId and not the actual query itself. can we not modify the interface itself and add `sqlQueryId(String id)`? The default impl of this method will be empty. That way, backward compatibility is maintained.

Reply:
Yes, we can augment the `QueryMetrics` interface itself since it is allowed in a major version release. I have added a `sqlQueryId(String)` method. Further, I've also cleaned up `GenericQueryMetricsFactory` wrt default implementation of the new `makeMetrics` method.
Thanks for the suggestion!